Assets/Scripts/Actors/ActorComponents/PlayerActorInventory.cs

```diff
@@ -88,29 +111,26 @@ public void CheckUseItem( RaycastHit hitInfo )
 
 	void CheckGiveResources( RaycastHit hitInfo )
 	{
-		if ( hitInfo.transform )
-		{
-			BuddyStats buddyStats = hitInfo.transform.GetComponent<BuddyStats>();
+		BuddyStats buddyStats = hitInfo.transform.GetComponent<BuddyStats>();
 
-			GodTag godTag = GetComponent<GodTag>(); // For checking if this actor owns the buddy
+		GodTag godTag = GetComponent<GodTag>(); // For checking if this actor owns the buddy
 
-			if ( buddyStats &&
-			     buddyStats.isAlive &&
-			     ( buddyStats.owner == null || buddyStats.owner == godTag ) )
-			{
-				buddyStats.SetGod( godTag );
-				GiveResource( buddyStats );
+		if ( buddyStats &&
+		     buddyStats.isAlive &&
+		     ( buddyStats.owner == null || buddyStats.owner == godTag ) )
+		{
+			buddyStats.owner = godTag;
+			GiveResource( buddyStats );
 
-				_playerActor.actorPhysics.OverrideLook(
-					buddyStats.GetComponent<Transform>().position - GetComponent<Transform>().position,
-					_lookOverrideDuration );
-			}
+			_playerActor.physics.OverrideLook(
+				buddyStats.GetComponent<Transform>().position - GetComponent<Transform>().position,
+				_lookOverrideDuration );
 		}
 	}
 
 	void GiveResource( BuddyStats buddyStats )
 	{
-		buddyStats.GiveResource( (actor as PlayerActor).actorStats, (ResourceData)heldResources[resourceIndex] );
+		buddyStats.GiveResource( (actor as PlayerActor).stats, (ResourceData)heldResources[resourceIndex] );
```

Comment:
Actually, if `actor` needs to be a `PlayerActor`, it should just be a `PlayerActor`. I think this problem stems from the `ActorComponent` base class, which should probably be removed.